syncopy/datatype/base_data.py

```diff
@@ -354,21 +354,48 @@ def _set_dataset_property_with_list(self, inData, propertyName, ndim):
             act = "real and complex NumPy arrays"
             raise SPYValueError(legal=lgl, varname="data", actual=act)
 
-        # Ensure shapes match up
-        if any(val.shape != inData[0].shape for val in inData):
-            lgl = "NumPy arrays of identical shape"
-            act = "NumPy arrays with differing shapes"
-            raise SPYValueError(legal=lgl, varname="data", actual=act)
+        # Requirements for input arrays differ wrt data-class (`DiscreteData` always 2D)
+        if any(["ContinuousData" in str(base) for base in self.__class__.__mro__]):
 
-        import ipdb; ipdb.set_trace()
+            # Ensure shapes match up
+            if any(val.shape != inData[0].shape for val in inData):
+                lgl = "NumPy arrays of identical shape"
+                act = "NumPy arrays with differing shapes"
+                raise SPYValueError(legal=lgl, varname="data", actual=act)
+            trialLens = [val.shape[self.dimord.index("time")] for val in inData]
 
-        # use stackingdim to build data-array!
-        if self.__class__.__name__ == "AnalogData":
-            pass # use np.h/vstack here
         else:
-            pass # allocate ndarray
-
 
+            # Ensure all arrays have shape `(N, 3)``
+            if any(val.shape[1] != 3 for val in inData):
+                lgl = "NumPy 2d-arrays with 3 columns"
+                act = "NumPy arrays of different shape"
+                raise SPYValueError(legal=lgl, varname="data", actual=act)
+            trialLens = [np.nanmax(val[:, self.dimord.index("sample")]) for val in inData]
+
+        # Now the shaky stuff: use determined trial lengths to cook up a (completely
+        # fictional) samplerate: we aim for `smax` Hz and round down to `sround` Hz
+        nTrials = len(trialLens)
+        sround = 50
+        smax = 1000
+        srate = min(max(min(smax, tlen / 2) // sround * sround, 1) for tlen in trialLens)
+        t0 = -srate
+        msg = "Artificially generated trial-layout: trigger offset = {t0} sec, " +\
+            "samplerate = {srate} Hz (rounded to {sround} Hz with max of {smax} Hz)"
+        SPYWarning(msg.format(t0=t0/srate, srate=srate, sround=sround, smax=smax), caller="data")
+
+        # Use constructed quantities to set up trial layout matrix
+        accumSamples = np.cumsum(trialLens)
```

Comment:
I don't wanna overcomplicate it.. still have to ask, wouldn't it be also somewhat easy to iterate over the individual numpy arrays and then just stack the respective trial-lengths such that also different trial lengths/nSamples could be realized? It's really nothing I think we would need right now, but if we wanna write tests with non-equal lengths trials I could imagine this to be very helpful as we can create more complicated data on the fly!

Reply:
Yes, absolutely. I just wanted to keep it simple for the time being. But this whole trial stacking can be extended to trials of different lengths - it just requires some add'l index gymnastics in for loops.

Reply:
Ok I created #170 to keep track of this :+1: